### ISSUE TITLE
Change unify output file format from .csv to .h5

### ISSUE THREAD

**Issue by tristan-ranff** [computational-ms/unify_idents#50]
Hdf files can be written and read a lot faster (e.g. msgfplus unify results):

```
start = datetime.datetime.now()
df.to_csv(kwargs["output_filenames"][0], index=False)
print((datetime.datetime.now() - start).total_seconds())
10.686126

start = datetime.datetime.now()
df = pd.read_csv(str(kwargs["output_filenames"][0]))
print((datetime.datetime.now() - start).total_seconds())
4.044992
```

```
start = datetime.datetime.now()
df.to_hdf(str(kwargs["output_filenames"][0]).replace(".csv", ".h5"), key="df", mode="w")
print((datetime.datetime.now() - start).total_seconds())
2.707328

start = datetime.datetime.now()
df = pd.read_hdf(str(kwargs["output_filenames"][0]).replace(".csv", ".h5"), key="df")
print((datetime.datetime.now() - start).total_seconds())
3.375205
```

Files are also slightly smaller:
```
❯ find . -maxdepth 1 -name "test_msgfplus_2021_03_22_unified*" ! -name "*.ujson" -exec ls -lh {} \;
378M Dec 16 15:20 ./test_msgfplus_2021_03_22_unified.csv
352M Dec 16 15:24 ./test_msgfplus_2021_03_22_unified.h5
```

In HDF5 empty strings would not be evaluated to None on pandas import in some columns which can be relevant for e.g. Modifications.

A disadvantage would be the readability of csv vs h5 files. Thoughts?

**Comment by ArtiVlasov** [comment]
Thinking on the bigger scale, h5 sounds promising. And even now during developing/debugging the files are partially huge, so its anyhow a pain to open those with excel or so to have a look. And if I anyhow import them somewhere, I don't care which read command to use.

so we could go for ugly and efficient, I would say :)

**Comment by StSchulze** [comment]
I agree that a lot of output files nowadays are too large to handle in Excel, so h5 sounds good to me.
Ideally the change would come with a simple node though, that can convert h5 into the "traditional" csv, for anyone who needs it.

As a side note, neither h5 nor csv are PSI formats, so it would be awesome to have a converter into mzTab or mzIdentML at some point in Ursgal 2, to also allow for full submissions to PRIDE, etc. ... but I'm pretty sure those would be larger and slower than h5/csv, so not really the point of this issue here ;)

**Comment by tristan-ranff** [comment]
> Ideally the change would come with a simple node though, that can convert h5 into the "traditional" csv, for anyone who needs it.

we could also add an output file type parameter with support for all pandas defaults and just use h5 as default value in uparma, so that by setting something like "unify_output_format" = "csv" you would still be able to generate csv's.

the converter sounds good as well.

**Comment by fu** [comment]
possible to see the same benchmark with parquet?
